package.json

```diff
@@ -15,17 +15,19 @@
     "test": "mocha --ui=tdd 'lib/**/*.test.js'"
   },
   "peerDependencies": {
+    "@supabase/supabase-js": ">=1.35.3",
```

Comment:
Actually, are you sure about the `peerDependencies` choice here?

All of these dependencies are directly referenced by this package. That doesn't seem like the use case `peerDependencies` was designed for. It seems to me these should be normal dependencies:

https://nodejs.org/en/blog/npm/peer-dependencies/

I think the difference is if replicache-todo doesn't depend on `pg-mem` will this all still compile? replicache-todo shouldn't need to depend on pg-mem, supabase, etc.

Reply:
As of npm 7 peer dependencies are auto installed https://github.blog/2021-02-02-npm-7-is-now-generally-available/#peer-dependencies, so replicache-todo doesn't need to install these.  After a ton of read, I believe peerDependencies is best for these specific dependencies.  The main difference is how conflicts are handled, with normal dependencies both versions are installed, which for some packages can lead to conflicts, and also leads to bundle size bloat.  With peerDependencies a suppressible error is raised.   Peer deps help avoid installing multiple versions of the same library.  But its debatable so I think it would be a good to have a discussion with the team once our npm expert @arv is back to decide on guidelines for when we use dependencies vs peerDependencies.

Reply:
OK thanks for thinking about this. I agree when @arv is back he should look at it. Can you put a task on your todo list to ask him when he returns?

Reply:
I'm hardly an expert here. I was trying to find a good guideline for when to use peerDeps vs deps. Earlier on it seemed that it was clear that you should only use peerDeps if there are no `require` in your code. But now it is very muddy.

To be honest I don't think there is any semantic difference any more. `dependencies` and `peerDependencies` are both added to node_modules and they both accept the same versioning syntax (i.e `"react": ">=16.0 <19.0"`).

Let's just use dependencies and make sure the version is as wide as possible.